lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -160,6 +161,38 @@ class BamDataResult(DataResult):
     max_high: int
 
 
+class DeleteDatasetBatchPayload(BaseModel):
+    datasets: List[DatasetSourceId] = Field(
+        description="The list of datasets IDs with their sources to be deleted/purged.",
+    )
+    purge: Optional[bool] = Field(
+        default=False,
+        description="Whether to permanently delete from disk the specified datasets."
+    )
+
+
+class DatasetErrorMessage(BaseModel):
+    dataset: DatasetSourceId = Field(
+        description="The encoded ID of the dataset and it's source.",
+    )
+    error_message: str = Field(
+        description="The error message returned while processing this dataset.",
+    )
+
+
+class DeleteDatasetBatchResult(BaseModel):
+    success_count: int = Field(
+        description="The number of datasets successfully processed.",
+    )
+    errors: Optional[List[DatasetErrorMessage]] = Field(
+        default=None,
+        description=(
+            "A collection of dataset IDs and the corresponding error message if something "
```

Comment:
```suggestion
            "A list of dataset IDs and the corresponding error message if something "
```
Just to reserve `collection` for dataset collections, even if that is clear here.